### PR TITLE
Add cross-compiling and scons 4 support for osx-arm64.

### PR DESCRIPTION
**Pull request by erykoff**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

I will be submitting these patches upstream.

One issue is that for some reason running `scons` with version 4 wasn't working, I had to call it via `python`.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -11,19 +11,25 @@ source:
...
+    - cross-python_{{ target_platform }}     # [build_platform != target_platform]
+    - {{ pin_compatible('numpy') }}                                  # [build_platform != target_platform]
     - {{ compiler('cxx') }}
-    - scons <4
+    - scons
 
   host:
     - python
-    - numpy
+    - {{ pin_compatible('numpy') }}
```
Comment (beckermr) [inline recipe/meta.yaml:32]: why did you change this line?

File: recipe/0002-bypass-fmath-non-intel-cpus.patch [hunk 2]
```diff
@@ -0,0 +1,43 @@
+--- a/include/galsim/fmath/fmath.hpp
++++ b/include/galsim/fmath/fmath.hpp
+@@ -33,6 +33,11 @@
```
Comment (beckermr) [inline recipe/0002-bypass-fmath-non-intel-cpus.patch:3]: @rmjarvis what are your thoughts on the performance hit from this patch?

Comment (erykoff) [reply recipe/0002-bypass-fmath-non-intel-cpus.patch:3]: He already approved it on galsim/main so ... not devastating?

File: recipe/meta.yaml (hunk 1 above)
Comment (erykoff) [reply recipe/meta.yaml:32]: This is what @isuruf told me to do but maybe it isn’t necessary here? I guess I haven’t tried without the change.

File: recipe/0002-bypass-fmath-non-intel-cpus.patch (hunk 2 above)
Comment (beckermr) [reply recipe/0002-bypass-fmath-non-intel-cpus.patch:3]: Ohh ok. Nvm then

File: recipe/meta.yaml (hunk 1 above)
Comment (beckermr) [reply recipe/meta.yaml:32]: I don't mind it I just don't understand it. I thought we needed numoy with no constraints in host for the global pinning to apply properly. 

 Have you checked the output packages?

Comment (beckermr) [reply recipe/meta.yaml:32]: Their lower numoy bounds should match the global pins on each platform and not the latest versions of numoy.

**Comment by beckermr** [comment]
@conda-forge-admin rerender

File: recipe/meta.yaml [hunk 3]
```diff
@@ -11,19 +11,25 @@ source:
...
+    - cross-python_{{ target_platform }}     # [build_platform != target_platform]
+    - {{ pin_compatible('numpy') }}          # [build_platform != target_platform]
     - {{ compiler('cxx') }}
-    - scons <4
+    - scons
 
   host:
     - python
-    - numpy
+    - numpy  # {{ pin_compatible('numpy') }}
```
Comment (beckermr) [inline recipe/meta.yaml:32]:
```suggestion
    - numpy
```

File: recipe/meta.yaml [hunk 4]
```diff
@@ -11,19 +11,25 @@ source:
...
 build:
-  number: 1
+  number: 2
   skip: true  # [win]
 
 requirements:
   build:
+    - python                                 # [build_platform != target_platform]
+    - cross-python_{{ target_platform }}     # [build_platform != target_platform]
+    - {{ pin_compatible('numpy') }}          # [build_platform != target_platform]
```
Comment (beckermr) [inline recipe/meta.yaml:26]:
```suggestion
    - numpy  # [build_platform != target_platform]
    # - {{ pin_compatible('numpy') }}          # [build_platform != target_platform]
```

**Comment by beckermr** [comment]
@erykoff can you rerender this please? The bot not being able to push to your forks is getting very annoying.

File: recipe/0002-bypass-fmath-non-intel-cpus.patch (hunk 2 above)
Comment (rmjarvis) [reply recipe/0002-bypass-fmath-non-intel-cpus.patch:3]: Right.  The fmath stuff requires SSE commands.  So if you have something that can't do SSE, then this just prevents it from breaking on compile.  In my experience the fmath stuff is like a 20-30% speed up for typical use cases, so that's about how much slower this will be than it could be if someone did the work to specialize the fast exp to systems that can't use SSE.  But that is presumably a fair bit of work to figure out.

File: recipe/meta.yaml (hunk 1 above)
Comment (beckermr) [reply recipe/meta.yaml]: This all appears to pass without the pin-compatible statements.

Comment (beckermr) [reply recipe/meta.yaml]: @isuruf can you comment on why you thought they were needed?

Comment (erykoff) [reply recipe/meta.yaml]: It may very well be I added this when having unrelated solving issues.

**Comment by erykoff** [comment]
@conda-forge-admin please rerender

**Comment by erykoff** [comment]
@beckermr In order to rerender I think somebody needs to fork mine.  I have no idea why, I'm not an organization and I haven't blocked anybody.  And clearly you are able to push changes to the PR...

**Comment by beckermr** [comment]
Can you do it locally?

**Comment by erykoff** [comment]
Ah, I guess I can!  And now I know how.

**Comment by erykoff** [comment]
I'm going to put in the patch from here https://patch-diff.githubusercontent.com/raw/GalSim-developers/GalSim/pull/1113.diff .

**Comment by erykoff** [comment]
@conda-forge/galsim This is ready.
